src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,64 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from 'react';
+import { SortBy } from '../../types/SortBy';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  query: string;
+  sortBy: SortBy;
+  handleChange: (query: string) => void;
+  handleSelect: (sortBy: SortBy) => void;
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: React.FC<Props> = ({
+  query,
+  handleChange,
+  sortBy,
+  handleSelect,
+}) => {
+  const handleResetQuery = () => {
+    handleChange('');
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            value={sortBy}
+            onChange={(event) => handleSelect(event.target.value as SortBy)}
```

Comment:
typescript is not pleased without `as` ?

Reply:
yes, Argument of type 'string' is not assignable to parameter of type 'SortBy'